### DISCUSSION TITLE
PiShrink creating files with different sizes with the same input

### DISCUSSION THREAD

**Discussion by jlueb** [Drewsif/PiShrink#263] [Q&A] [answered]
Hi, I've recently started using PiShrink to decrease the size of backup images and I noticed that the file size of an with PiShrink shrunken and xz compressed image differs if I repeat it with the same input image.
But in comparison to that if I only shrink the image with PiShrink and then manually compress it with xz it's always the same size.
Shouldn't PiShrink always produce the same result with the same input? Does someone know the reason why that's not the case?

**Comment by Drewsif** [comment, marked as answer]
So in theory, yes it should. But PiShrink is calling a lot of external tools that I don't know for sure will always produce the same output given the same input. Its very likely they are not and that is why you are seeing a size difference. But they should end roughly around the same size.
